src/services/ethereum/transaction.ts

```diff
@@ -206,29 +196,32 @@ export class TransactionExecutor implements ITransactionExecutor {
     logger.debug(`${scheduled._id} Condition: asset=${scheduled.conditionAsset} amount=${scheduled.conditionAmount}`);
 
     let currentConditionAmount;
-
     try {
       const token = new ethers.Contract(transaction.to, abi, provider);
-      currentConditionAmount = (await token.balanceOf(transaction.from)) as BigNumber;
+      logger.debug(` transaction.from: ${transaction.from} sched: ${scheduled._id}  `);
+      currentConditionAmount = (await token.balanceOf(transaction.from)) as BigNumber; // always zero
+      logger.debug(`currentConditionAmount1: ${currentConditionAmount}`);
     } catch (e) {
-      currentConditionAmount = await provider.getBalance(transaction.from!);
+      provider.getBalance(transaction.from).then((balance) => {
+        currentConditionAmount = ethers.utils.formatEther(balance);
```

Comment:
`formatEther` changes wei to eth which will totally mess up the logic that depends on wei everywhere else